metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -2012,6 +2012,20 @@ class MetalsLanguageServer(
               .asScala
           } yield ().asInstanceOf[Object]
         }
+      case ServerCommands.ExtractMethod(
+            ServerCommands.ExtractMethodParams(doc, range, defnPos)
+          ) =>
+        CancelTokens.future { token =>
+          val uri = doc.getUri()
+          for {
+            edits <- compilers.extractMethod(doc, range, defnPos, token)
+            if (!edits.isEmpty())
```

Comment:
Let's use  `logging.logErrorWhen`, otherwise we will get an unclear error with filter.